heron/metricscachemgr/src/java/com/twitter/heron/metricscachemgr/metricscache/MetricsCacheQueryUtils.java

```diff
@@ -0,0 +1,290 @@
+// Copyright 2016 Twitter. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.twitter.heron.metricscachemgr.metricscache;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.logging.Logger;
+
+import com.twitter.heron.metricscachemgr.metricscache.datapoint.ExceptionDatapoint;
+import com.twitter.heron.proto.system.Common;
+import com.twitter.heron.proto.tmaster.TopologyMaster;
+
+/**
+ * define the query request and response format
+ */
+public final class MetricsCacheQueryUtils {
+  // logger
+  private static final Logger LOG = Logger.getLogger(MetricsCacheQueryUtils.class.getName());
+
+  private MetricsCacheQueryUtils() {
+  }
+
+  /**
+   * compatible with com.twitter.heron.proto.tmaster.TopologyMaster.MetricRequest
+   */
+  public static MetricRequest Convert(TopologyMaster.MetricRequest request) {
```

Comment:
If TopologyMaster.MetricRequest is a protobuf and MetricRequest is not, better to name method `fromProtobuf` so it's clear from the name what kind of conversion is happening. Same for other methods int this class